### ISSUE TITLE
Implement sampling vertices from subtractions

### ISSUE THREAD

**Issue by gipert** [legend-exp/remage#56]
Useful to simulate e.g. K42 decays outside mini-shroud but inside cylinder.

**Comment by gipert** [comment]
I'm not sure what I meant here, but this seems to me already supported.

**Comment by ManuelHu** [comment]
I suppose you meant "from a substraction of two user-defined solids" (i.e. where none of them is from the original geometry). This is not yet possible

**Comment by gipert** [comment]
Ah yes you are right...

**Comment by tdixon97** [comment]
do we really need this? whats the application? E..g LAr inside MS can be simulated already

**Comment by gipert** [comment]
LAr outside MS?

**Comment by tdixon97** [comment]
Yes I get it now what you want is to select points where A & !B compared to what you have now which is A& B. Although thats slightly different to what Manuel suggested above.
Do we really want to add more features to the RMG generators though, or move to python entirely? For me its seems much easier, and better to do it sooner.

**Comment by gipert** [comment]
In C++ we have `::IsInside()`, that's quite crucial...

**Comment by tdixon97** [comment]
Ok now I am remembering this discussion. I will work on this implementation and meanwhile think of some high level tests we can do.
